### ISSUE TITLE
Wrong type for `get_lr` inside lr_scheduler.pyi

### ISSUE THREAD

**Issue by Dipet** [pytorch/pytorch#100804]
### 🐛 Describe the bug

`lr_scheduler.pyi` describes `def get_lr(self) -> float: ...` but the implementation returns `List[float]`

https://github.com/pytorch/pytorch/blob/31f311a816c026bbfca622d6121d6a7fab44260d/torch/optim/lr_scheduler.pyi#L19
https://github.com/pytorch/pytorch/blob/31f311a816c026bbfca622d6121d6a7fab44260d/torch/optim/lr_scheduler.py#L394

### Versions

Current master branch

cc @vincentqb @jbschlosser @albanD @janeyx99 @ezyang @malfet @rgommers @xuzhao9 @gramster

**Comment by janeyx99** [comment]
A brief glance at the lr schedulers does incriminate our get_lr() to be a lie in terms of type but also intention-->it looks like get_lr() really gets the lr's per param group, but promises just one value.

We would accept a preliminary fix that would:
- [ ] Document correctly the intention of get_lr(). It looks like a bunch of the impls already redirect to get_last_lr already.
- [ ] Rectify the type in the .pyi file (hopefully this is not BC breaking but I have unfortunate sneaking suspicions)

The above can be multiple PRs.

A better definitely bc breaking fix would:
- [ ] Figure out the right APIs to offer in LR scheduler
- [ ] Implement and do BC breaking things to rectify the API. There's no current maintainer for lr schedulers, so this option will not be easily committed to.

**Comment by wbittles** [comment]
If we look at the  lr_scheduler.pyi , we can see that the type hint annotations for 2 methods are incorrect.

As discussed above get_lr method is declared as returning a float when in fact it returns a list of floats
However get_last_lr , returns a float, yet the type hint annotation reports a list of floats

`   def get_last_lr(self) -> List[float]: ...
    def get_lr(self) -> float: ...`  

I traced the call stack for get_last_lr and the code does expect a single float value, and the implementation return a single float.
I also trace the call stack and implementation of get_lr and this code also expects a list of floats.

so the obvious solution seems to be that these type hint annotation have been switched in error and simply flipping the annotations  back seems to address the problem.

As the changes are trivial I've opened a PR to try and close this issue.

**Comment by jewelltaylor** [comment]
Still facing this issue now. Would be happy to put up a PR but see that #104957 was not merged in. Is there any reason?

**Comment by janeyx99** [comment]
@jewelltaylor lr_scheduler.pyi should have been merged into the same file as lr_scheduler.py now, so this issue should be fixed. Please reopen if something is still not corrected.

And for posterity, the reason that PR wasn't merged was because get_last_lr also returns a list and not a float.
